file_catalog/mongo.py

```diff
@@ -2,25 +2,35 @@
 
 import datetime
 import logging
+from concurrent.futures import ThreadPoolExecutor
+
+import pymongo
+from bson.objectid import ObjectId
+from pymongo import MongoClient
+from pymongo.errors import BulkWriteError
+from tornado.concurrent import run_on_executor
+
 try:
     from collections.abc import Iterable
 except ImportError:
     from collections import Iterable
 
-import pymongo
-from pymongo import MongoClient
-from pymongo.errors import BulkWriteError
-from bson.objectid import ObjectId
 
-from concurrent.futures import ThreadPoolExecutor
-from tornado.concurrent import run_on_executor
 
 logger = logging.getLogger('mongo')
 
 class Mongo(object):
     """A ThreadPoolExecutor-based MongoDB client"""
-    def __init__(self, host=None, port=None, authSource=None, username=None, password=None):
-        logger.info('MongoClient args: host=%s, port=%s, username=%s', host, port, username)
+    def __init__(self, host=None, port=None, authSource=None, username=None, password=None, uri=None):
+
+        if uri:
+            logger.info(f"MongoClient args: uri={uri}")
+            self.client = MongoClient(uri, authSource=authSource).file_catalog
+        else:
+            logger.info('MongoClient args: host=%s, port=%s, username=%s', host, port, username)
+            self.client = MongoClient(host=host, port=port,
+                                      authSource=authSource,
+                                      username=username, password=password).file_catalog
 
         self.client = MongoClient(host=host, port=port,
                                   authSource=authSource,
```

Comment:
Does the line override what was set in the if/else block above?

Reply:
It looks like it does.  That seems like a copy/paste error?